source/controlTypes.py

```diff
@@ -711,3 +742,38 @@ def processNegativeStates(role, states, reason, negativeStates):
 		# This is not a state change; only positive states were supplied.
 		# Return all negative states which should be spoken, excluding the positive states.
 		return speakNegatives - states
+
+def processAndLabelStates(role, states, reason, positiveStates, negativeStates, positiveStateLabelDict={}, negativeStateLabelDict={}):

```

Comment:
In another PR @jcsteh cautioned against using empty dicts as default function parameters. I wonder if that applies here.